src/status_im/contexts/wallet/wallet_connect/events/session_requests.cljs

```diff
@@ -95,21 +95,23 @@
                      :transaction  tx
                      :chain-id     chain-id
                      :display-data display-data)
-      :fx [[:dispatch [:wallet-connect/show-request-modal]]]})))
+      :fx [(when-not refetch? [:dispatch [:wallet-connect/show-request-modal]])]})))
 
 (rf/reg-event-fx
  :wallet-connect/process-eth-send-transaction
- (fn [{:keys [db]}]
+ (fn [{:keys [db]} [refetch?]]
```

Comment:
I'd suggest to not reuse this event by passing the `refetch?` flag to avoid re-showing the modal (in general flags like this should be avoided in most cases), but instead to only re-use the `:effects.wallet-connect/prepare-transaction`. Since we also need the logic from `:wallet-connect/prepare-transaction-success` (but without the "show modal" dispatch), it can be extracted into a separate event and used for both preparing and updating the tx.

Reply:
If i understand it correctly you mean to create a new event for update transaction like this? 

so `:wallet-connect/prepare-transaction-success` still get a flag to skip the modal part or not. right? 

```clojure
(rf/reg-event-fx
 :wallet-connect/update-transaction
 (fn [{:keys [db]}]
   (let [event    (data-store/get-db-current-request-event db)
         tx       (-> event data-store/get-request-params first)
         chain-id (-> event
                      (get-in [:params :chainId])
                      networks/eip155->chain-id)]
     (when tx
       {:fx [[:effects.wallet-connect/prepare-transaction
              {:tx         tx
               :chain-id   chain-id
               :on-success #(rf/dispatch [:wallet-connect/prepare-transaction-success % chain-id true])
               :on-error   #(rf/dispatch [:wallet-connect/on-processing-error %])}]]}))))
```

Reply:
So this is one way, but another way would be to have separate events for the success handlers, but still share the logic, either with a common event for updating the state but without showing the modal, or just a utility function. The the success event can be passed to the `prepare-transaction` event, or alternatively have different events for preparing and updating the transaction.

This is just off the top of my head, so I might be missing things, but the point is that there are ways to avoid passing the flag, even if we end up with a few more events. Passing the flag through two events just to avoid showing the modal will lead to other issues in the future. If you're stuck, we can go through it in a call at some point and figure it out together.

Reply:
Honestly, I don’t see any issue with extending a function by adding an option to reuse it, as long as it doesn’t cause breaking changes. This approach is better than duplicating logic under different event names. I’ve noticed similar cases where the function content is identical, but only the event name differs, which isn’t ideal. Extending a function still aligns with the Open/Closed Principle and allows us to reuse logic without duplication. 

wdyt? @shivekkhurana @ilmotta

Reply:
I'm confused, if can't pass an argument to a func to extend it, how we can avoid duplication, maybe if you can give me some example that would be nice.

Reply:
I don't have any strong preference guys, either way can work well and has pros/cons.

Side note? the solution in `:wallet-connect/prepare-transaction-success` to dispatch `:wallet-connect/show-request-modal` is more of a UI concern. This is not a problem from this PR or in general recent PRs, but a thing we tend to do quite often in the code base, more than we should perhaps. This could be one of the reasons why it's difficult to model some of our events, because they are caring about UI concerns that the view should decide.

Reply:
we could extract the logic from the events into an external function that would be used by the events to avoid duplicating the logic, but since there's no strong preferences, I won't push on this

Reply:
Without an example, it’s hard to fully grasp the implementation you’re suggesting, but I’ll try to see if I can use the logic without adding flags or causing duplication.

Reply:
Thanks for your feedback @clauxx, I applied some changes, could you please recheck?